airflow/providers/apache/kafka/hooks/kafka_consumer.py

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from kafka import KafkaConsumer
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class KafkaConsumerHook(BaseHook):
+    """KafkaConsumerHook Class."""
+
+    DEFAULT_HOST = 'kafka1'
+    DEFAULT_PORT = 9092
+
+    def __init__(self, topic, host=DEFAULT_HOST, port=DEFAULT_PORT, kafka_conn_id='kafka_default'):
+        super().__init__(None)
+        self.conn_id = kafka_conn_id
+        self._conn = None
+        self.server = None
+        self.consumer = None
+        self.extra_dejson = {}
+        self.topic = topic
+        self.host = host
+        self.port = port
+
+    def get_conn(self) -> KafkaConsumer:
+        """
+            A Kafka Consumer object.
+
+        :return:
+            A Kafka Consumer object.
+        """
+        if not self.consumer:
+            conn = self.get_connection(self.conn_id)
+            service_options = conn.extra_dejson
+            host = conn.host or self.DEFAULT_HOST
+            port = conn.port or self.DEFAULT_PORT
+
+            self.server = f"""{host}:{port}"""
+            self.consumer = KafkaConsumer(self.topic, bootstrap_servers=self.server, **service_options)
+        return self.consumer
+
+    def get_messages(self, timeout_ms=5000) -> dict:
+        """
+        Get all the messages haven't been consumed, it doesn't
```

Comment:
>  "Get all the messages haven't been consumed,"

If we use poll() without max_records, the behavior is returns at most "max_poll_records" #records. "max_poll_records" is setted to 500 by default at Consumer Init config.

So, we're not going to consume "all" message except we put a very high number as max_poll_records(could be a memory bomb) or we have a low number of message in the topic.

https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.poll